### DISCUSSION TITLE
Request information is lost in POST from client component

### DISCUSSION THREAD

**Discussion by miguelisidoro** [remix-run/remix#10117] [Q&A] [answered]
Hi,

I am passing the following information to an action in Remix.

{"url":"www.microsoft.com"}

The code is the following:

```
const url = `/gistinformation/submit`;
 const request = { url: params.userInput };
 const requestAsString = JSON.stringify(request);

const response = await fetch(aiAssistantSubmitActionUrl, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: requestAsString,
      });
```

I confimed that {"url":"www.microsoft.com"} is being passed to the server.
When the information reaches the Remix action the request is empty. The action code:

```
export async function action({ request }: any) {
const requestAsString = JSON.stringify(request.json());
    console.log("Request: " + requestAsString);
    const response = await getResponse(request);
    console.log("Response: " + response);
}
```

requestAsString has the following value in it:{}, an empty json object.

Did this happened to anyone? What I am doing wrong?

PS: This code was working fine today and suddently stopped working after I did a code refactoring in the server side part (node.js), not in the client side code or the action code.

Thanks

**Comment by sergiodxa** [comment, marked as answer]
You need to add `await` before `request.json()`, otherwise you're stringifying a Promise

**Reply by miguelisidoro** [reply]
Thanks, that was it!
